### PR TITLE
simply logic inside MasterBaseTaskExecThread.submit method

### PR DESCRIPTION
**Pull request by gabrywu**
## What is the purpose of the pull request

simply logic inside MasterBaseTaskExecThread.submit method

## Brief change log

  - replace complex while phrase with two for phrase inside  MasterBaseTaskExecThread.java

### CODE REVIEW

**Comment by gabrywu** [comment]
Could anybody review this PR ？and give some advice?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java [hunk 1]
```diff
@@ -97,82 +100,85 @@ public MasterBaseTaskExecThread(TaskInstance taskInstance){
...
+                if (task != null && task.getId() != 0) {
+                    submitDB = true;
+                } else {
+                    logger.error("task commit to db failed , taskId {} has already retry {} times, please check the database",
+                            taskInstance.getId(), retryTimes);
+                    Thread.sleep(commitRetryInterval);
                 }
-                if(submitDB && !submitTask){
+            } catch (Exception e) {
+                logger.error("task commit to mysql and dispatcht task failed", e);
```
Comment (Rubik-W) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java:140]: It is recommended to optimize the log content, add task information.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java [hunk 2]
```diff
@@ -97,82 +100,85 @@ public MasterBaseTaskExecThread(TaskInstance taskInstance){
...
+            for (int retryTimes = 1; retryTimes <= commitRetryTimes && !submitTask; retryTimes++) {
+                try {
                     // dispatch task
                     submitTask = dispatchTask(task);
+                    if (!submitTask) {
+                        logger.error("task commit  failed , taskId {} has already retry {} times, please check", taskInstance.getId(), retryTimes);
+                        Thread.sleep(commitRetryInterval);
+                    }
+                } catch (Exception e) {
+                    logger.error("task commit to mysql and dispatcht task failed", e);
```
Comment (Rubik-W) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java:154]: It is recommended to optimize the log content, add task information.
"not mysql"

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java (hunk 1 above)
Comment (gabrywu) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java]: cool, good idea. added

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java (hunk 2 above)
Comment (gabrywu) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java]: cool, good idea. added

**Comment by gabrywu** [comment]
> The probability of failure here is low, so it is recommended to merge the retry logic

yes,you are right. I'm planning to simply the logic first, and then refactor these codes with RetryerUtils

**Comment by qiaozhanwei** [comment]
suggest use RetryerUtils refactor this code

because I can't find 'for' is better 'while' in  readability
Thx

**Comment by gabrywu** [comment]
> suggest use RetryerUtils refactor this code
> 
> because I can't find 'for' is better 'while' in readability
> Thx

nothing to do with for and while. original logic is a lot complex, I just want to keep it simple.
I will create another PR using RetryerUtils

**Pull request closed without merging**